### PR TITLE
fs-2175-rescore-section-display

### PR DESCRIPTION
**Pull request by ramsharma-prog**

### CODE REVIEW

File: app/assess/routes.py [hunk 1]
```diff
@@ -110,14 +111,21 @@ def display_sub_criteria(
...
+        score_form = ScoreForm()
+        rescore_form = RescoreForm()
+        score_error = (
+            justification_error
+        ) = scores_submitted = is_rescore = False
         if request.method == "POST":
             current_app.logger.info(f"Processing POST to {request.path}.")
-            if form.validate_on_submit():
-                score = int(form.score.data)
+            if rescore_form.validate_on_submit():
```
Comment (ramsharma-prog) [inline app/assess/routes.py:121]: can we add some comments as there are multiple forms?

File: app/assess/templates/assessor_tasklist.html [hunk 2]
```diff
@@ -16,8 +16,10 @@
 
 <div class="govuk-phase-banner flex-parent-element">
     <p class="govuk-!-text-align-left flex-parent-element flexed-element-margins-collapse">
-        {{ govukBackLink({'href': url_for("assess_bp.landing"), 'text': 'Back to assessment dashboard'}) }}
+        <a href="{{url_for('assess_bp.landing')}}" class="govuk-back-link" id="back-to-assessment-dashboard">
+            Back to <b>assessment dashboard</b></a>
```
Comment (tferns) [inline app/assess/templates/assessor_tasklist.html:20]: Any reason from switching off the partial? not against it cause it gives us more flexibility but could've done:
```suggestion
        {{ govukBackLink({'href': url_for("assess_bp.landing"), 'html': 'Back to <b>assessment dashboard</b>', 'attributes': {'id': 'back-to-assessment-dashboard'}}) }}
```

Comment (tferns) [reply app/assess/templates/assessor_tasklist.html:20]: We also have this on 4 pages, so I'd either:

- make sure these changes are applied to all locations
- refactor out into a partial and change all locations them to use that

The two locations not on this PR but on main are:
https://github.com/communitiesuk/funding-service-design-assessment/blob/c548325b1ad08d2640d955be21c3be91b5eb5362/app/assess/templates/continue_assessment.html#L11

https://github.com/communitiesuk/funding-service-design-assessment/blob/c548325b1ad08d2640d955be21c3be91b5eb5362/app/assess/templates/resolve_flag.html#L11

**Review by tferns — CHANGES_REQUESTED** [review]

**Review by tferns — COMMENTED** [review]
Let me know if you want to talk through any my prior comments, if we need to get this bug-fix out quickly we can talk about raising stories to fix it up later.  but preference would be to tackle them now in case they never get done

File: app/assess/routes.py [hunk 3]
```diff
@@ -126,11 +134,10 @@ def display_sub_criteria(
                     sub_criteria_id=sub_criteria_id,
                 )
                 scores_submitted = True
```
[the remark this answers is not in the document]
Comment (tferns) [reply app/assess/routes.py:136]: Here's an example of flashing I'd implemented before: https://github.com/communitiesuk/funding-service-design-assessment/pull/124

File: app/assess/routes.py (hunk 1 above)
Comment (gio-karunakaran) [reply app/assess/routes.py:121]: have added doc-strings in form classes and comment in this line also 👍🏾

File: app/assess/routes.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (gio-karunakaran) [reply app/assess/routes.py]: score_submitted removed and not needed 👍🏾

File: app/assess/routes.py [hunk 4]
```diff
@@ -126,11 +134,10 @@ def display_sub_criteria(
...
-
-            else:
-                score_error = True if not form.score.data else False
+                is_rescore = False
+                score_error = True if not score_form.score.data else False
                 justification_error = (
-                    True if not form.justification.data else False
+                    True if not score_form.justification.data else False
                 )
```
[the remark this answers is not in the document]
Comment (gio-karunakaran) [reply app/assess/routes.py]: score error/ justification error are no longer used 👍🏾 (using flask form errors now)

File: app/assess/routes.py [hunk 5]
```diff
@@ -110,14 +111,21 @@ def display_sub_criteria(
             abort(404)
         # SECURITY SECTION END ######
```
[the remark this answers is not in the document]
Comment (gio-karunakaran) [reply app/assess/routes.py:113]: simplified score/ rescore logic 👍🏾

**Review by tferns — APPROVED** [review]
Thanks for working through those changes with me!

**Pull request merged**